### PR TITLE
2FA major update

### PR DESCRIPTION
**Pull request by Rick-Anderson**
@TomArcher @rustd please review

### CODE REVIEW

**Comment by TomArcher** [comment]
@Erikre Please review

File: aspnet/security/2fa.rst [hunk 1]
```diff
@@ -1,133 +1,109 @@
...
 
-Running the Application
------------------------
+.. literalinclude:: 2fa/sample/WebSMS/src/WebSMS/Services/MessageServices.cs
+   :language: c#
+   :lines: 12-39
+   :dedent: 4 
 
-Run the application and after a quick build step, you should see it open in your web browser.
+.. comment note:: Twilio cannot target dnxcore50: You will get compilation errors if you build your project when dnxcore50 is included because Twilio does not have a package for dnxcore50. You can remove dnxcore50 from the *project.json* file or you can call the Twilio REST API to send SMS messages.
```
Comment (rustd) [inline aspnet/security/2fa.rst]: You can point to this blog to get Twilio running on .NET Core https://www.twilio.com/blog/2015/08/getting-started-with-asp-net-5-and-visual-studio-code-on-a-mac.html

File: aspnet/security/2fa/sample/WebSMS/src/WebSMS/Services/MessageServices.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        public Task SendSmsAsync(string number, string message)
+        {
+            var twilio = new Twilio.TwilioRestClient(
+                Options.SID,           // Account Sid from dashboard
+                Options.AuthToken);    // Auth Token
+
+            var result = twilio.SendMessage(Options.SendNumber, number, message);
+            // Use the debug output for testing without receiving a SMS message.
+            // Remove the Debug.WriteLine(message) line after debugging.
+            System.Diagnostics.Debug.WriteLine(message);
```
Comment (rustd) [inline aspnet/security/2fa/sample/WebSMS/src/WebSMS/Services/MessageServices.cs]: We need to use Logging Factory here to log messages.

File: aspnet/security/2fa.rst [hunk 3]
```diff
@@ -135,55 +111,62 @@ Let us run the Web site and show the two-factor authentication flow.
...
-        options.Lockout.MaxFailedAccessAttempts = 10;
-    });
+Debugging  Twilio
+-------------------
+If you're able to use the Twilio API, but you don't get an SMS message, try the following:
 
+1. Log in to the Twilio site and navigate to the **Logs** > **SMS & MMS Logs** page. You can verify that messages were sent and delivered.
+2. Use the following code in a console application to test Twilio:
 
-Next steps
```
Comment (rustd) [inline aspnet/security/2fa.rst]: Why is this deleted? Users need to know how to set these secrets in Azure Web Apps

File: aspnet/security/2fa/sample/WebSMS/src/WebSMS/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                .AddDefaultTokenProviders();
+
+            // Add MVC services to the services container.
+            services.AddMvc();
+
+            // Uncomment the following line to add Web API services which makes it easier to port Web API 2 controllers.
+            // You will also need to add the Microsoft.AspNet.Mvc.WebApiCompatShim package to the 'dependencies' section of project.json.
+            // services.AddWebApiConventions(); 
+
+            services.Configure<Microsoft.AspNet.Identity.IdentityOptions>(options =>
```
Comment (rustd) [inline aspnet/security/2fa/sample/WebSMS/src/WebSMS/Startup.cs]: Add a using for this namespace and remove it from here

**Comment by rustd** [comment]
Once you make these changes then we are good

**Comment by danroth27** [comment]
@Rick-Anderson Did you mean to include the  2fa/sample/sampleB7 folder? I think the entire sampleB7 folder should be removed.

**Comment by danroth27** [comment]
Also, remove `MessageService - Copy.cs` in the Services folder. I don't think it's used anywhere.

**Comment by danroth27** [comment]
:watch:

File: aspnet/security/2fa.rst [hunk 5]
```diff
@@ -1,189 +1,171 @@
...
+ - Create a class to fetch the secure SMS key. For this sample, the ``AuthMessageSMSSenderOptions`` class is created in the *Services/AuthMessageSMSSenderOptions.cs* file.
 
- - Add code in MessageServices to send SMS
+.. literalinclude:: 2fa/sample/WebSMS/src/WebSMS/Services/AuthMessageSMSSenderOptions.cs
+   :language: c#
+   :lines: 3-8
+   :dedent: 4
 
-.. code-block:: c#
+Set ``SID``, ``AuthToken``, and ``SendNumber`` with the `secret-manager tool <http://docs.asp.net/en/latest/security/app-secrets.html>`_. For example:
```
Comment (TomArcher) [inline aspnet/security/2fa.rst]: Recommend being more explicit about setting these values to the values obtained from Twilio.

**Comment by Rick-Anderson** [comment]
@danroth27  removed both

**Comment by Rick-Anderson** [comment]
@rustd @danroth27  addressed everything, need squirrel.

**Comment by Rick-Anderson** [comment]
@rustd @danroth27 addressed everything, need squirrel.

**Comment by danroth27** [comment]
:shipit:
